src/kili/mutations/label/__init__.py

```diff
@@ -154,61 +155,49 @@ def append_to_labels(
         result = self.auth.client.execute(GQL_APPEND_TO_LABELS, variables)
         return format_result("data", result, Label)
 
-    @typechecked
-    def append_labels(
-        self,
-        json_response_array: List[dict],
-        author_id_array: Optional[List[str]] = None,
-        label_asset_external_id_array: Optional[List[str]] = None,
-        label_asset_id_array: Optional[List[str]] = None,
-        seconds_to_label_array: Optional[List[int]] = None,
-        label_type: str = "DEFAULT",
-        project_id: Optional[str] = None,
-    ):
-        """Append a label to an asset.
+    def append_labels(self, labels: List[AppendLabelData], label_type: LabelType = "DEFAULT"):
+        """Append labels to assets.
 
-        Args:
-            json_response_array: Labels to upload
-            author_id_array: IDs of the author of the labels
-            label_asset_external_id_array: External identifier of the assets
-                on which to add the labels
-            label_asset_id_array: Identifier of the assets on which to add the labels
-            seconds_to_label_array: Time to create the labels
-            project_id: Identifier of the project
-            label_type: Can be one of `DEFAULT`, `PREDICTION`, `REVIEW` or `INFERENCE`
+        !!! info "fields of AppendLabelData"
+            ```
+            class AppendLabelData(TypedDict, total=False):
+                asset_id: Required[str]
+                json_response: Required[Dict]
+                author_id: str
+                seconds_to_label: int
+                modelName: str
+            ```
 
-        !!! warning
-            Either provide `label_asset_id` or `label_asset_external_id` and `project_id`
+        Args:
+            labels: list of dictionnaries with informations about the labels to create.
+            label_type: Can be one of `AUTOSAVE`, `DEFAULT`, `PREDICTION`, `REVIEW` or `INFERENCE`
 
         Returns:
             A result object which indicates if the mutation was successful,
                 or an error message.
 
         Examples:
             >>> kili.append_to_labels(
-                label_asset_id_array=[asset_id1, asset_id2],
-                json_response=[{...}, {...}]
+                    [
+                        {'json_response': {...}, asset_id: 'cl9wmlkuc00050qsz6ut39g8h'},
```

Comment:
This is a "V3"-style syntax, are we sure we should push it? No pb to use it behind the API, but imho we should remain consistent with the other methods.

Reply:
Even the resolver in the backend now has the style of array of dict instead of several arrays of fields. So if people want to use the resolver directly in graphQL, they will have to send the labels as array of dict. It is indeed not in line with other methods but for me we should then begin to switch, don't you think ? No conversion needed in the function and easier to use for the user.